File: LeetCode_Marathons/LCM2/Medium/Count_Primes/a.py
class Solution:
    def countPrimes(self, n: int) -> int:
        isNotPrimeStatuses = [False] * n 
        counter = 0

        for i in range(2, n):
            if not isNotPrimeStatuses[i]:
                if n ** 0.5 >= i:
                    for j in range(i * i, n, i): isNotPrimeStatuses[j] = True
                counter += 1

        return counter

File: LeetCode_Marathons/LCM2/Medium/Count_Primes/a.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<bool> isNotPrimeStatuses(n + 1, false);
        int counter = 0;
        for (int i = 2; i < n; i++) {
            if (!isNotPrimeStatuses[i]) {
                if (sqrt(n) >= i) {
                    for (int j = i * i; j < n; j += i) isNotPrimeStatuses[j] = true;
                }
                counter++;
            }
        }
        return counter;
    }
};